src/backend/impl/src/repositories/types/proposal.rs

```diff
@@ -114,43 +118,28 @@ impl TryFrom<ProposalInfo> for Proposal {
     fn try_from(nns_proposal: ProposalInfo) -> Result<Self, Self::Error> {
         let proposal_id = nns_proposal
             .id
-            .ok_or(ApiError::internal("Proposal id is None"))?
+            .ok_or(ApiError::internal(
+                "Failed to map NNS proposal: Proposal id is None",
+            ))?
             .id;
 
-        let inner_proposal = nns_proposal
-            .clone()
-            .proposal
-            .ok_or(ApiError::internal("Inner proposal is None"))?;
-
-        let title = inner_proposal
-            .title
-            .unwrap_or_else(|| String::from("Title not available"));
-
-        let topic = nns_proposal.topic().try_into()?;
-        let nervous_system = NervousSystem::new_network(proposal_id, topic);
+        let nervous_system = NervousSystem::new_network(proposal_id, nns_proposal.clone());
 
         let state = nns_proposal.status().into();
```

Comment:
Sorry I must've missed this in a previous PR, but we shouldn't be taking the `state` from the proposal.

`state` refers to the review period state, while `status` refers to the voting period.

Reply:
The `From<ProposalStatus>` just sets the state of the proposal in our canister to `InProgress` if the proposal's status is `Open`, otherwise it sets it to `Completed`.
It was just a way to "safely" set the state of the polled proposal. Shouldn't we leave it like it is?

Reply:
Fixed after the discussion we had yesterday in person